### PR TITLE
Configuration and time service to protobuf.

### PR DESCRIPTION
**Pull request by dvush**
## Overview

---

### Definition of Done

- [x] There are no TODOs left in the merged code
- [x] Change is covered by automated tests
- [x] Benchmark results are attached (if applicable)
- [x] The [coding guidelines] are followed
- [x] Public API has proper documentation
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: exonum/src/encoding/protobuf/mod.rs [hunk 1]
```diff
@@ -216,6 +216,16 @@ impl ProtobufConvert for u64 {
...
 
+impl ProtobufConvert for i32 {
+    type ProtoStruct = i32;
+    fn to_pb(&self) -> Self::ProtoStruct {
+        *self
+    }
+    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, ()> {
+        Ok(pb)
+    }
+}
```
Comment (ilammy) [inline exonum/src/encoding/protobuf/mod.rs]: I think it makes sense now to just add

```rust
macro_rules! trivial_protobuf_type {
    ($T:ty) => {
        impl ProtobufConvert for $T {
            type ProtoStruct = $T;
            fn to_pb(&self) -> Self::ProtoStruct {
                *self
            }
            fn from_pb(pb: Self::ProtoStruct) -> Result<Self, ()> {
                Ok(pb)
            }
        }
    }
}
```

already and implement `ProtobufConvert` for all standard integer types.

Comment (dvush) [reply exonum/src/encoding/protobuf/mod.rs]: Yes, that was discussed already. I agree.

Comment (aleksuss) [reply exonum/src/encoding/protobuf/mod.rs]: @dvush add this in separate PR ?

File: services/configuration/Cargo.toml [hunk 2]
```diff
@@ -27,8 +28,12 @@ serde_derive = "1.0.10"
 serde_json = "1.0.19"
 tempdir = "=0.3.7"
 toml = "=0.4.9"
+protobuf = "=2.2.0"
```
Comment (aleksuss) [inline services/configuration/Cargo.toml]: `=` is redundant here ?

**Pull request merged**